File: 11/js/1.js
const fs = require('fs');
const { createMonkey, monkeysToString, playRound } = require("./Monkey");

(function () {
  const contents = fs.readFileSync("../input.txt").toString();
  const notes = contents.split("\r\n\r\n").filter(x => x.length > 0).map(m => m.split("\r\n").filter(x => x.length > 0));

  const monkeys = notes.map(createMonkey);
  const T = x => Math.floor(x / 3);
  for (let i = 0; i < 20; i++) playRound(monkeys, T);

  console.log("===== ITEMS =====")
  console.log(monkeysToString(monkeys));

  // Print times inspected
  console.log("===== COUNT =====");
  monkeys.forEach((mk, i) => console.log(`Monkey ${i} inspected items ${mk.count} times.`));
  const counts = monkeys.map(mk => mk.count);
  const max1 = Math.max(...counts);
  counts.splice(counts.indexOf(max1), 1);
  const max2 = Math.max(...counts);
  console.log(`\nMost active: ${max1} and ${max2}\nMonkey Business: ${max1 * max2}`);
})();

File: 11/hs/1.hs
import System.IO
import Data.List
import Data.List.Split
import Monkey

main = do
  h <- openFile "../input.txt" ReadMode
  contents <- hGetContents h
  let
    notes = map (tail.lines) $ splitOn "\n\n" contents
    mks = map create_monkey notes
    mks' = process_n_monkeys 20 mks (Just (`div` 3))
    work = map get_count mks'
    max1 = maximum work
    max2 = maximum $ delete max1 work
  putStrLn $ to_str mks'
  putStr "\n"
  print work
  print $ "Most: " ++ (show max1) ++ " and " ++ (show max2)
  print $ "Monkey Business: " ++ (show $ max1 * max2)
  hClose h